docs/README.md

```diff
@@ -993,25 +993,11 @@ export class CartPaid {
 }
 ```
 
-An event has to know the ID of the entity it belongs to, and there are several strategies to do so. One would be injecting the entity ID directly in the constructor, or as a nested attribute. Alternatively, for events like `ProductCreated` it is common to return a brand new ID as the entity did not exist. For _singleton_ entities, where there's only one instance, you can even use a constant value. 
+An event has to know the ID of the entity it belongs to, that's why you have to implement the `entityID` method. For getting the entity ID, you could inject it directly in the event's constructor, or as a nested attribute. If your domain requires a _singleton_ entity, where there's only one instance, you can return a constant value. In the `CartPaid` example, the entity ID (`paymentID`) is injected directly.
 
-In the `CartPaid` example, the entity ID (`paymentID`) is injected directly, and here is another example of a newly generated value:
-
-```typescript
-@Event
-export class ProductCreated {
-  public constructor(
-    readonly displayName: string,
-    readonly price: Money
-  ) {}
+Note:
 
-  public entityID(): UUID {
-    // returns a new UUID because the Product entity
-    // does not exist yet
-    return UUID.generate()
-  }
-}
-```
+> The `entityID` method must always return the same value for the same event's instance because Booster will call it several times during its lifecycle. Otherwise, the entities' data will corrupt.
```

Comment:
I would avoid saying things like `or your data will be corrupted`, as it could frighten people. Also, people might find clever ways of returning different IDs for the same event instance (not sure how). 
Here is a suggestion following what I have seen in other docs when you use something as it is not intended to be used:
```suggestion
> The `entityID` method must always return the same value for the same event's instance. Otherwise, the result of the entity reduction will be undefined.
```

Reply:
Would this phrasing make more sense?

```suggestion
> The `entityID` method must always return the same value for the same event's instance. Otherwise, the result of the entity reduction will be unpredictable.
```

Reply:
I like it better. `undefined` sounds bad too

Reply:
Agree